### ISSUE TITLE
xenial: nuget certificate validation error (since 20170410)

### ISSUE THREAD

**Issue by tahina-pro** [tianon/docker-brew-ubuntu-core#86]
Consider the following Dockerfile:
```
FROM ubuntu:xenial
RUN apt-get --yes update
RUN apt-get --yes install --no-install-recommends ca-certificates-mono nuget
RUN nuget update -self
```

With `ubuntu:xenial-20170214`, building an image from this Dockerfile succeeds.

With `ubuntu:xenial-20170410` and `ubuntu:xenial-20170417.1`, building now fails, with the following error:
Error: SendFailure (Error writing headers)

When adding the official Mono repositories as follows, per instructions given at http://www.mono-project.com/docs/getting-started/install/linux/:
```
RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
RUN echo "deb http://download.mono-project.com/repo/debian wheezy main" > /etc/apt/sources.list.d/mono-xamarin.list
```

the image successfully builds with `ubuntu:xenial-20170214`, but fails with `ubuntu:xenial-20170410` and `ubuntu:xenial-20170417.1` with the following error message:
Error: SecureChannelFailure (Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED)

(Of course such failures arise not only with `nuget update -self`, but also when trying to install any package using `nuget`.)

Other uses of HTTPS (e.g. `wget`, `git`) seem to succeed with those Ubuntu images.

**Comment by graingert** [comment]
```Docker
FROM ubuntu:xenial-20170417.1
RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
RUN echo "deb http://download.mono-project.com/repo/debian wheezy main" > /etc/apt/sources.list.d/mono-xamarin.list
RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates-mono nuget
RUN apt-get update && apt-get install -y --no-install-recommends curl
RUN cert-sync /etc/ssl/certs/ca-certificates.crt
RUN nuget update -self
```

fails for me too.

**Comment by tianon** [comment]
This looks like this is possibly related to https://github.com/fsprojects/docker-fsharp/issues/24 -- regardless, this doesn't seem like there's anything we can do at the image level. :disappointed:

If it's a bug in a package in Ubuntu, it ought to be filed in launchpad (Ubuntu's bugtracker), but it _looks_ like it's something in Mono itself. :confused:

**Comment by tahina-pro** [comment]
I think I found a solution -- or, at least, the following fix works for me: install the `tzdata` package.
```
apt-get install tzdata
```

Diagnosis: when I was trying to run the TLS Hello World from http://www.mono-project.com/docs/getting-started/mono-basics/, I got the following error:
```
Unhandled Exception:
System.IO.FileNotFoundException: Could not find file "/etc/localtime"
```
Then, I checked and indeed, `/etc/localtime` was pointing to `/usr/share/zoneinfo/Etc/UTC`, which did not exist until I installed `tzdata`.

So, it seems that `tzdata` was removed from the default Ubuntu image. At least that's what I understood.

Hopefully this can be useful to anyone experienced similar issues.
